### ISSUE TITLE
Duplicated logs from disk to server

### ISSUE THREAD

**Issue by melix-sun** [rsyslog/rsyslog#5411]
Hi,

we use disk-Assisted In-memory Queue to sysnc log to cloud server. we found one issue that some parts of log are always uploaded repeatedly. 
I have attached the configuration ,is anyone also have the same issue for duplicated logs to cloud server from disk?

**Comment by rgerhards** [comment]
Note that rsyslog follows a "at last once" principle. Duplication can e.g. happen when the transmission is interrupted. This is often the case when load balancers are involved.

**Comment by melix-sun** [comment]
> Note that rsyslog follows a "at last once" principle. Duplication can e.g. happen when the transmission is interrupted. This is often the case when load balancers are involved.

Hi rgerhards,

Thanks for your reply. but it seems that the duplicated logs are always the same once restart the client, and the logs are always kept in disk even though they had been uploaded to cloud server sucessfully.

**Comment by rgerhards** [comment]
Do you have any error messages from rsyslog? This sounds like either a problem with the queue files or some blocked access to state files etc. rsyslog usually complains in these cases. so I suggest to review logs.

**Comment by rgerhards** [comment]
you could also create a debug log, at least from the startup. Maybe this helps us better understand what is happening. But I would suggest to check the rsyslog logs first (less work to do for everyone).

Instructions for debug log: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html

**Be sure that the problem in question occurs at least once during the debug run**. Once this is done, please post the debug log (e.g. via pastebin or github). After that, we can review the contents and check how exactly rsyslog processing worked.

**Comment by melix-sun** [comment]
> you could also create a debug log, at least from the startup. Maybe this helps us better understand what is happening. But I would suggest to check the rsyslog logs first (less work to do for everyone).
> 
> Instructions for debug log: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html
> 
> **Be sure that the problem in question occurs at least once during the debug run**. Once this is done, please post the debug log (e.g. via pastebin or github). After that, we can review the contents and check how exactly rsyslog processing worked.

Hi rgerhards，

Thanks， we will check the logs first and create debug log if need ,will give you more detail after that.

**Comment by melix-sun** [comment]
Hi rgerhards,

we check the log again ,and the details please find the attachment of the configuration, thanks. As for the debug file , it seems that I cannot upload the debug file to this site. You can tell me how can I share the debug file if you know, thanks
[Duplication details.xlsx](https://github.com/user-attachments/files/16278364/Duplication.details.xlsx)

**Comment by rgerhards** [comment]
Apologies for the very late reply. This is probably no longer relevant to the original reporter, but I am adding the conclusion for people who arrive here via search while we work through the backlog. We do not close rsyslog issues lightly, and we are not closing this merely because of age.

Triage result: duplicate / superseded.

Reasoning: This earlier duplicate-log report was superseded by #5419 from the same reporter with fuller configuration, debug data, and discussion.

If this still reproduces on a current rsyslog release and maintained platform, please open a new issue with the current version, complete configuration, logs, and a minimal reproducer so we can investigate it as a fresh current problem.
